src/FetchAuthTokenCache.php

```diff
@@ -142,6 +142,14 @@ public function signBlob($stringToSign, $forceOpenSsl = false)
             );
         }
 
+        // Pass the access token from cache to GCECredentials for signing a blob.
+        // This saves a call to the metadata server when a cached token exists.
+        if ($this->fetcher instanceof Credentials\GCECredentials) {
+            $cached = $this->fetchAuthTokenFromCache();
+            $accessToken = isset($cached['access_token']) ? $cached['access_token'] : null;
+            return $this->fetcher->signBlob($stringToSign, $forceOpenSsl, $accessToken);
+        }
+
         return $this->fetcher->signBlob($stringToSign, $forceOpenSsl);
```

Comment:
```suggestion
        $accessToken = null;
        if ($this->fetcher instanceof Credentials\GCECredentials) {
            $cached = $this->fetchAuthTokenFromCache();
            $accessToken = isset($cached['access_token']) ? $cached['access_token'] : null;
        }

        return $this->fetcher->signBlob($stringToSign, $forceOpenSsl, $accessToken);
```

Just the tiniest of nits, feel free to totally ignore this.

Reply:
I am going to ignore this, but only because the _only_ fetcher which accepts the third argument is `GCECredentials`, and so I think it's misleading to pass the third argument in, even if null, to the other fetchers.